visidata/loaders/xlsx.py

```diff
@@ -53,6 +53,61 @@ def iterload(self):
         for rownum in Progress(range(worksheet.nrows)):
             yield list(worksheet.cell(rownum, colnum).value for colnum in range(worksheet.ncols))
 
+            
+@VisiData.api
+def save_xlsx(vd, p, *sheets):
+    import openpyxl
+
+    wb = openpyxl.Workbook()
+    wb.remove_sheet(wb['Sheet'])
+
+    for vs in sheets:
+        ws = wb.create_sheet(title=vs.name)
+
+        headers = [col.name for col in vs.visibleCols]
+        ws.append(headers)
+
+        for dispvals in vs.iterdispvals(format=False):
+
+            row = []
+            for v in dispvals.values():
+                if type(v) == date:
+                    v = datetime.datetime.fromtimestamp(int(v.timestamp()))
+                elif not type(v) in [int,float,str]:
+                    v = str(v)
+                row.append(v)
+            ws.append(row)
+
+    wb.active = ws
+
+    wb.save(filename=p)
+    status(f'{p} save finished')
+
+
+@VisiData.api
+def save_xls(vd, p, *sheets):
+    import xlwt
+
+    wb = xlwt.Workbook()
+
+    for vs in sheets:
+
+        ws1 = wb.add_sheet(vs.name[:32]) # sheet name can not be longer than 32 characters
+
+        for col_i, col in enumerate(vs.visibleCols):
+            ws1.write(0, col_i, col.name)
+
+        for r_i, dispvals in enumerate(vs.iterdispvals(format=False)):
+            r_i += 1
+            for c_i, v in enumerate(dispvals.values()):
+
+                if not type(v) in [int,float,str]:
```

Comment:
I think this should be `isNumeric(col)`, which would also detect currency, date, and vlen types.  `visidata.date` inherits from `datetime.datetime` btw, so did you try using it directly instead of doing any conversions?

Reply:
Hey, yea, I tried giving it `date` directly but unfortunately openpyxl looks at the `type` when deciding if it is a object it can consume and convert to an Excel type. In this case it's looking for datetime.datetime.

I'll have a look and try to test with `isNumeric`. I have a feeling it'll break currency formatting....!